test/unit/connection_string.test.ts

```diff
@@ -89,6 +94,37 @@ describe('Connection String', function () {
     expect(options.credentials.source).to.equal('$external');
   });
 
+  it('should retain user-specified authSource with no credentials', function () {
```

Comment:
in what way are the assertions below "retaining" the authSource? the previous assertions were about having the property present, but now it's actually omitting it, so I'm confused about the expected behavior here. What's supposed to happen to authSource=someDb?

Reply:
I changed the functionality after the test was written whoops! the credentials object should be discarded if the only auth related option specified is authSource. Updated!